Fix Android Studio Warnings in InstantEditorViewModel.kt

VictoriaTaiwan opened the pull request:
## Purpose / Description
- Made vals punctuationAtStart, capturedWord and punctuationAtEnd not-nullable.
- Replaced unnecessary safe calls with dot calls.
- Replaced "get" calls with indexing operator.

## Fixes
* Fixes #13282

## Approach
Used Inspect Code feature in Android Studio to detect existing warnings in the project and used suggested by Android Studio fixes for the issues in the InstantEditorViewModel file.

## How Has This Been Tested?
Tested the changes on emulator in Android Studio.
- SDK version - API level 34

## Learning (optional, can help others)
Official emulation guide is helpful for testing code changes. Please refer to this page https://developer.android.com/studio/run/emulator.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

criticalAY reviewed (approved):
LGTM! thanks

david-allison commented:
Cheers!

david-allison commented:
@VictoriaTaiwan You have your `git config` set to expose your email publicly

* `git.name` = `I****`
* `git.email` = `b**e@gmail.com`

Plus, the email is not associated with your GitHub, so you won't get credit on your GitHub heatmap

If you want to change this, please change your settings, rebase the PR and force push the changes so we can merge this.

If you're happy with this data being kept in the repo git history, please let us know

I would personally recommend using an "ID based noreply" address for GitHub: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

VictoriaTaiwan commented:
@david-allison Thank you for mentioning this issue and reviewing my PR! I changed my git config, rebased the PR and force pushed the changes. Please let me know if something still needs changes.

mikehardy reviewed (approved):
Nice! Thank you

Arthur-Milchior commented:
Hi @VictoriaTaiwan
I see you just uploaded your proposal on GSoC website. Please join our discord if you want to get feedback on it. Right now, it's not enough details for us to be able to consider it. You did some work for us, we'd be happy to explain how to improve it. But usually we do it on discord and I can't see you there
